plugins/woocommerce-blocks/tests/e2e/tests/stock-filter/stock-filter.block_theme.spec.ts

```diff
@@ -0,0 +1,288 @@
+				name: 'Apply',
+			} )
+		).toBeVisible();
+	} );
+} );
+
+test.describe( `${ blockData.name } Block - with PHP classic template`, () => {
+	test.beforeEach( async ( { admin, page, editor } ) => {
+		await cli(
+			'npm run wp-env run tests-cli -- wp option update wc_blocks_use_blockified_product_grid_block_as_template false'
+		);
+
+		await admin.visitSiteEditor( {
+			postId: 'woocommerce/woocommerce//archive-product',
+			postType: 'wp_template',
+		} );
+
+		await editor.canvas.locator( 'body' ).click();
+
+		await editor.insertBlock( {
+			name: 'woocommerce/filter-wrapper',
+			attributes: {
+				filterType: 'stock-filter',
+				heading: 'Filter By Price',
+			},
+		} );
+		await editor.saveSiteEditorEntities();
+		await page.goto( `/shop` );
+	} );
+
+	test( 'should show all products', async ( { frontendUtils } ) => {
+		const legacyTemplate = await frontendUtils.getBlockByName(
+			'woocommerce/legacy-template'
+		);
+
+		const stockFilter = await frontendUtils.getBlockByName(
+			'woocommerce/filter-wrapper'
+		);
+
+		const products = legacyTemplate
+			.getByRole( 'list' )
+			.locator( '.product' );
+
+		await expect( products ).toHaveCount( 16 );
+
+		await expect( stockFilter.getByText( 'In Stock' ) ).toBeVisible();
+		await expect( stockFilter.getByText( 'Out of Stock' ) ).toBeVisible();
+	} );
+
+	test( 'should show only products that match the filter', async ( {
+		frontendUtils,
+	} ) => {
+		const stockFilter = await frontendUtils.getBlockByName(
+			'woocommerce/filter-wrapper'
+		);
+
+		await stockFilter.getByText( 'Out of Stock' ).click();
+
+		const legacyTemplate = await frontendUtils.getBlockByName(
+			'woocommerce/legacy-template'
+		);
+
+		const products = legacyTemplate
+			.getByRole( 'list' )
+			.locator( '.product' );
+
+		await expect( products ).toHaveCount( 1 );
+	} );
+} );
+
+test.describe( `${ blockData.name } Block - with Product Collection`, () => {
+	test.beforeEach(
+		async ( {
+			admin,
+			editorUtils,
+			productCollectionPageObject,
+			editor,
+		} ) => {
+			await admin.createNewPost();
+			await productCollectionPageObject.insertProductCollection();
+			await productCollectionPageObject.chooseCollectionInPost(
+				'productCatalog'
+			);
+			await editor.insertBlock( {
+				name: 'woocommerce/filter-wrapper',
+				attributes: {
+					filterType: 'stock-filter',
+					heading: 'Filter By Stock',
+				},
+			} );
+			await editorUtils.publishAndVisitPost();
+		}
+	);
+
+	test( 'should show all products', async ( { page } ) => {
+		const products = page
+			.locator( '.wp-block-woocommerce-product-template' )
+			.getByRole( 'listitem' );
+
+		await expect( products ).toHaveCount( 9 );
+	} );
+
+	test( 'should show only products that match the filter', async ( {
+		page,
+	} ) => {
+		await page.getByText( 'Out of Stock' ).click();
+
+		await page.waitForURL( ( url ) =>
+			url
+				.toString()
+				.includes( blockData.urlSearchParamWhenFilterIsApplied )
+		);
```

Comment:
Are we waiting for a navigation change here, or do we want to ensure the query params are applied? If it's the latter, let's use the [toHaveURL](https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-url) assertion instead:

```js
await expect( page ).toHaveURL( new RegExp( blockData.urlSearchParamWhenFilterIsApplied ) );